### PR TITLE
Add action buttons to the update view page.

### PR DESCRIPTION
**Pull request by ralphbean**
This includes allowing the editing of updates in 2f93ff9, which is essential.

This also adds "revoke" as a possible action on a request which got dropped in
the port from bodhi1 on accident (added back in in e5c248a).

### CODE REVIEW

File: bodhi/models/models.py [hunk 1]
```diff
@@ -977,7 +982,12 @@ def set_request(self, action, request, pathcheck=True):
             notifications.publish(topic=topic, msg=dict(
                 update=self, agent=request.user.name))
             return
-
+        elif action is UpdateRequest.revoke:
+            self.revoke()
+            flash_log("%s has been obsoleted" % self.title)
```
Comment (lmacken) [inline bodhi/models/models.py]: s/obsoleted/revoked/

File: bodhi/models/models.py [hunk 2]
```diff
@@ -1448,12 +1458,42 @@ def unpush(self):
...
+                                 "one that is %s" % self.status.description)
+
+        # Remove the 'pending' koji tags from this update so taskotron stops
+        # evalulating them.
+        if self.request is UpdateRequest.testing:
+            self.remove_tag(self.release.pending_testing_tag)
+        elif self.request is UpdateRequest.stable:
+            self.remove_tag(self.release.pending_stable_tag)
+
+        # If we're revoking a request we need to set the status back to what it
```
Comment (lmacken) [inline bodhi/models/models.py]: Revoking a 'stable' request on an update with the 'testing' status won't revert the status back to pending. I don't think we need to change any enums other than set the `request` to `None`.

File: bodhi/models/models.py [hunk 3]
```diff
@@ -1448,12 +1458,42 @@ def unpush(self):
...
+        if self.release.testing_tag in current_tags:
+            self.status = UpdateRequest.testing
+        elif self.release.stable_tag in current_tags:
+            self.status = UpdateRequest.stable
+        else:
+            self.status = None
+
+        self.request = None
+
+        mail.send_admin('revoked', self)
```
Comment (lmacken) [inline bodhi/models/models.py]: With FMN I'm not sure we need to send mails for most of the triggers in `bodhi/mail.py`. What do you think?

Comment (ralphbean) [inline bodhi/models/models.py]: :+1:  I was just keeping the pattern in place that was already there (obsolete sent a mail, so revoke should too).

File: bodhi/models/models.py (hunk 2 above)
Comment (ralphbean) [inline bodhi/models/models.py]: Ah, the trick I think is that when the request got previously set to ``stable``, that code *also* sets the ``status`` to ``pending``, I think.  I guess I should double-check that.

Comment (ralphbean) [inline bodhi/models/models.py]: Nope.  I was mistaken.  I'll remove this code.  Thanks!

Comment (lmacken) [inline bodhi/models/models.py]: The `status` gets updated & the` request` gets unset towards the end of the push process by the `masher https://github.com/fedora-infra/bodhi/blob/pyramid/bodhi/masher.py#L242

File: bodhi/models/models.py [hunk 4]
```diff
@@ -1448,12 +1458,31 @@ def unpush(self):
...
 
         self.pushed = False
         self.status = UpdateStatus.unpushed
         mail.send_admin('unpushed', self)
 
+    def revoke(self):
+        """ Remove pending request for this update """
+        log.debug("Revoking %s" % self.title)
+
+        if self.status is not UpdateStatus.pending:
```
Comment (lmacken) [inline bodhi/models/models.py]: A `testing` update should probably be able to revoke a `stable` request.

Comment (ralphbean) [inline bodhi/models/models.py]: Agreed.

**Comment by lmacken** [comment]
:+1:

**Pull request merged**